weed/iamapi/iamapi_management_handlers.go

```diff
@@ -309,35 +313,42 @@ func (iama *IamApiServer) DeleteUserPolicy(s3cfg *iam_pb.S3ApiConfiguration, val
 
 func GetActions(policy *PolicyDocument) (actions []string) {
 	for _, statement := range policy.Statement {
+		glog.V(3).Infof("parsing statement: %+v", statement)
 		if statement.Effect != "Allow" {
 			continue
 		}
 		for _, resource := range statement.Resource {
+			glog.V(3).Infof("parsing resource: %s", resource)
 			// Parse "arn:aws:s3:::my-bucket/shared/*"
 			res := strings.Split(resource, ":")
+			glog.V(3).Infof("parsed resource: %v", res)
 			if len(res) != 6 || res[0] != "arn" || res[1] != "aws" || res[2] != "s3" {
-				glog.Infof("not match resource: %s", res)
+				glog.Infof("not a valid resource: %s", res)
 				continue
 			}
 			for _, action := range statement.Action {
+				glog.V(3).Infof("parsing action: %s", action)
 				// Parse "s3:Get*"
 				act := strings.Split(action, ":")
+				glog.V(3).Infof("parsed action: %v", act)
 				if len(act) != 2 || act[0] != "s3" {
-					glog.Infof("not match action: %s", act)
+					glog.Infof("not a valid action: %s", act)
 					continue
 				}
 				statementAction := MapToStatementAction(act[1])
-				if res[5] == "*" {
+				glog.V(3).Infof("parsed statementAction: %s", statementAction)
+				path := res[5]
+				if path == "*" {
 					actions = append(actions, statementAction)
 					continue
 				}
 				// Parse my-bucket/shared/*
-				path := strings.Split(res[5], "/")
-				if len(path) != 2 || path[1] != "*" {
-					glog.Infof("not match bucket: %s", path)
-					continue
-				}
-				actions = append(actions, fmt.Sprintf("%s:%s", statementAction, path[0]))
```

Comment:
With this, we only allowed a bucket name to be provided, not a path.
Now we allow any path.